.github/workflows/maven.yaml

```diff
@@ -110,6 +109,12 @@ jobs:
       run: mvn -B -V -e -ntp "-Dstyle.color=always" -DskipFormat ${{ matrix.profile.args }}
       env:
         MAVEN_OPTS: -Djansi.force=true
+    - name: Build with Maven (ZooKeeper compatibility build)
+      if: ${{ github.base_ref != 'main' }}
```

Comment:
Why exclude main?

Reply:
The reason for this change is that #3200 uses newer ZooKeeper API objects, so this check fails. #3200 isn't merged, but if it is, then this is an issue.

Reply:
Okay, that's a compelling reason to bump the minimum. What's the minimum for it to work? Can we just modify the one existing line, rather than do the changes proposed in this PR that jumps through hoops to check for main?

Reply:
I'll look.

Reply:
According to [this](https://github.com/apache/zookeeper/commit/553639378d5cf86c2488afff4586e5e4cce38061), it looks like 3.6.0 might be the minimum.

Reply:
I will modify the GitHub action file in my branch and see if that resolves it

Reply:
If that's the case, then changing it to 3.6.4 (the latest 3.6 bugfix) would probably be appropriate.

Reply:
3.6.1 was the minimum. I updated it in my PR, I'm going to close this.

Reply:
3.6.1 is the minimum from an API perspective. Do you think the API changed between 3.6.1 and 3.6.4 that would cause an issue, but still works with 3.8.1?

Reply:
I hope not. That would be uncharacteristically unstable of the ZK team.